### PR TITLE
Add terratest maven plugin and new category to docs

### PR DESCRIPTION
**Pull request by szgergo**
GitHub issue: https://github.com/gruntwork-io/terratest/issues/821
I have added terratest maven plugin to the docs as agreed.

### CODE REVIEW

File: docs/_docs/02_testing-best-practices/tools-and-plugins.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  Additional tools and plugins for terratest to make integration with existing workflows easier.
+tags: ["testing-best-practices", "alternatives","plugins","tooling"]
+order: 214
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+## Tools and Plugins
+
+Since almost all project has an infrasturcture that needs to be tested, it is great to have different kind of tools and plugins which can help integrating Terratest into your build pipelines seamlessly. Here is a quick list of the different tooling and various plugins out there.
```
Comment (brikis98) [inline docs/_docs/02_testing-best-practices/tools-and-plugins.md:15]:
```suggestion
This page contains a list of tools and plugins for Terratest to make integration with existing workflows easier. If you've created other tools that integrate with Terratest, a PR to add it to this list is very welcome!

1. [Terratest Maven Plugin](#terratest-maven-plugin)
```

File: docs/_docs/02_testing-best-practices/tools-and-plugins.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Terratest Maven Plugin
+
+Terratest Maven Plugin aims to bring the marvelous terratest to the JVM world. Create your Go based tests beside your Java code with Maven and run them together. You can export the results into Json or an HTML page. As the plugin is MIT licenced it is easy and painless to integrate into any Java+Maven combination. To learn more check out the website: [Terratest Maven Plugin](https://terratest-maven-plugin.github.io) and the [GitHub repository](https://github.com/terratest-maven-plugin/terratest-maven-plugin)
```
Comment (brikis98) [inline docs/_docs/02_testing-best-practices/tools-and-plugins.md:19]:
```suggestion
The Terratest Maven Plugin aims to bring Terratest to the JVM world. Create your Go based tests beside your Java code with Maven and run them together. You can export the results into Json or an HTML page. As the plugin is MIT licensed, it is easy and painless to integrate into any Java+Maven combination. To learn more check out the website: [Terratest Maven Plugin](https://terratest-maven-plugin.github.io) and the [GitHub repository](https://github.com/terratest-maven-plugin/terratest-maven-plugin)
```

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR!

File: docs/_docs/02_testing-best-practices/tools-and-plugins.md (hunk 2 above)
Comment (szgergo) [reply docs/_docs/02_testing-best-practices/tools-and-plugins.md]: Hi! Thanks! Did the changes, can you please check them again?

**Review by brikis98 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**